scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -1513,7 +1513,9 @@ package internal.trees {
     def rank: Int
     def tree: Tree
     def pt: Class[_]
-    def become[T <: Tree: AstInfo]: T with Quasi
+    def become[T <: Tree: AstInfo]: T
+    // `become` should be returning `T with Quasi`,
+    // workaround for https://github.com/scala/scala3/issues/22177
```

Comment:
The workaround in question. We could also add another method, which we then would use instead in ScalametaParser, keeping the more type safe `become` for the users (or at least Scala 2 users).

Reply:
it's under `internal`, so should be fine. otherwise, you'd get mima errors.

Reply:
It might, but requires way more changes (e.g. adding with Quotes to where AstInfo is generated, and specifying `with Quotes` for every typeBound where we pass Tree).